subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -225,6 +229,24 @@ include::sample[dir="buildCache/cacheable-bundle/groovy",files="build.gradle[tag
 include::sample[dir="buildCache/cacheable-bundle/kotlin",files="build.gradle.kts[tags=bundle-task]"]
 ====
 
+=== Using annotations
+
+Instead of using the runtime API, it is actually better to create a custom task class which is cacheable using <<more_about_tasks.adoc#sec:task_input_output_annotations,annotations to declare the inputs and outputs>>.
+
+We continue the example from above, trying to get a cacheable bundle task.
+Given that we already have the `NpmTask`, we now create a subclass declaring the inputs and outputs.
+
+When possible, it is better to use delegation instead of creating a subclass.
```

Comment:
Isn't this only possible for `Exec` and `JavaExec` themselves? What other tasks are there where delegation is an option?

Reply:
It is also possible for `copy` and `sync.

Reply:
OK, so we mention two possibilities out of four. I think it would be better to mention all of these explicitly.

Reply:
I now mention all of those.

For third party tasks, there may be helper classes one can use instead of calling the task action. I mentioned the built-in tasks as examples what ships with Gradle.